src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -79,6 +82,64 @@ public void remove(Person toRemove) {
         }
     }
 
+    /**
+     * Replaces the specified {@code target} with {@code editedTag} for all contacts.
+     */
+    public void setContactTag(Tag target, Tag editedTag) {
+        requireAllNonNull(target, editedTag);
+        int count = internalList.size();
+        for (int i = 0; i < count; i++) {
+            Person original = internalList.get(i);
+            if (original.getTags().contains(target)) {
```

Comment:
Would it be better to initialise a new HashSet with the original tags, remove the target, then add the editedTag in order to prevent arrowhead-style code? Same can be done for the `removeContactTag`